### DISCUSSION TITLE
Set counter via Berry Script with a variable

### DISCUSSION THREAD

**Discussion by Salvoitaly2023** [arendst/Tasmota#22512] [Q&A]
Hello dear community,

I would like to set a counter via Berry Script, where the value of the counter is in a variable.

So if I enter the following:

tasmota.cmd(‘counter1 50’)

it works fine and the counter is set to 50.

But if I want something like this:

var number = 50
tasmota.cmd(‘counter1’, number)

then it doesn't work and the counter is set to 4294967197.

What am I doing wrong? How can I include a variable in the command?
Unfortunately, I haven't found anything on the Internet about this.
I hope someone can help me please.

Thank you very much

**Comment by sfromis** [comment]
What you are doing wrong is not reading the syntax of `tasmota.cmd` (don't scour the Internet, read the Tasmota [docs for Berry](https://tasmota.github.io/docs/Berry/)); that function takes a **string** as first argument, and an optional second argument with a bool for logging or not.

What could be used in your case is `tasmota.cmd(f'Counter1 {number}')` or without using f-strings also `tasmota.cmd('Counter1 ' + str(number))` - both ways are building a single string with command and argument.

**However**, both ways are not the most direct approach to setting a counter from Berry, as the [`gpio` module](https://tasmota.github.io/docs/Berry/#gpio-module) has the function `counter_set`, meaning that your example would be:
```
import gpio
gpio.counter_set(0, number)
```
In addition to being faster, this also works in places where you can't use `tasmota.cmd`

**Comment by Salvoitaly2023** [comment]
Hello sfromis,

Yes I had read the Tasmota docs too, but overlooked the one with the string.
But your example with 

       import gpio
       gpio.counter_set(0, number)

is of course much better and I have also used it in my script.

Thank you very much!
